### PR TITLE
doc: openthread: Adding documentation about TCAT.

### PR DESCRIPTION
**Pull request by canisLupus1313**
This commit add TCAT documentation section in NCS Thread related docs.

### CODE REVIEW

File: doc/nrf/protocols/thread/tcat.rst  [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+such as inside a ceiling or embedded in a wall. Instead of scanning physical install codes once installed devices are powered,
+Thread commissioning can be implemented over authenticated TLS (Transport Layer Security) by exchanging security certificates
+This authenticated commissioning can be performed while in proximity using a mobile device (phone, tablet) over a wireless connection
+such as a Bluetooth Low Energy radio.
+
+Building and Running
+********************
+
+Device site
+===========
```
Comment (maciejbaczmanski) [inline doc/nrf/protocols/thread/tcat.rst :16]:
```suggestion
===========

```

File: doc/nrf/protocols/thread/tcat.rst  [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+Currently NCS provides experimental support for TCAT. Complie CLI sample with TCAT snippet enabled. :ref:`ot_cli_sample_activating_variants`
+
+After flashing the device this feature can be anabled in following way using CLI.
+    .. code-block:: console
+        ot tcat start
+
+Commissioner site
+=================
```
Comment (maciejbaczmanski) [inline doc/nrf/protocols/thread/tcat.rst :24]:
```suggestion
=================

```

File: doc/nrf/protocols/thread/tcat.rst  [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+Currently only available comissioner tool can be found in Openthread repository. Please refer to instruction on how to use it:
```
Comment (maciejbaczmanski) [inline doc/nrf/protocols/thread/tcat.rst :25]:
```suggestion
Currently the only available commissioner tool can be found in OpenThread repository. Please refer to instruction on how to use it:
```

**Review by maciejbaczmanski — COMMENTED** [review]
I'm not sure, but I think each sentence in `.rst` file should be written in one line, with next sentences in next lines

**Comment by canisLupus1313** [comment]
@wiba-nordic This is a new way of performing commissioning, since it has significant differences I would prefer to keep it separate.

File: doc/nrf/protocols/thread/tcat.rst [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+Thread commissioning can be implemented over authenticated TLS (Transport Layer Security) by exchanging security certificates.
+This authenticated commissioning can be performed while in proximity, using a mobile device (phone, tablet) over a wireless connection such as a Bluetooth Low Energy radio.
+
+Building and Running
+********************
+
+Device site
+===========
+
+Currently NCS provides experimental support for TCAT. Compile CLI sample with TCAT snippet enabled: :ref:`ot_cli_sample_activating_variants`
```
Comment (wiba-nordic) [inline doc/nrf/protocols/thread/tcat.rst:18]:
```suggestion
The |NCS| currently provides experimental support for TCAT.
To test this feature, build the :ref:`ot_cli_sample` sample with the :ref:`TCAT snippet <ot_cli_sample_activating_variants>` enabled.
```

File: doc/nrf/protocols/thread/tcat.rst [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+After flashing the device this feature can be anabled in following way using CLI.
+
+    .. code-block:: console
+        uart:~$ ot tcat start
+
```
Comment (wiba-nordic) [inline doc/nrf/protocols/thread/tcat.rst:24]:
```suggestion
After flashing the sample to the device, use the following command to enable TCAT:

.. code-block:: console
   uart:~$ ot tcat start

```

File: doc/nrf/protocols/thread/tcat.rst [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+Commissioner site
+=================
```
Comment (wiba-nordic) [inline doc/nrf/protocols/thread/tcat.rst:26]:
```suggestion
```

File: doc/nrf/protocols/thread/tcat.rst [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+Currently the only available commissioner tool can be found in OpenThread repository. Please refer to instruction on how to use it:
+https://github.com/nrfconnect/sdk-openthread/tree/main/tools/tcat_ble_client
```
Comment (wiba-nordic) [inline doc/nrf/protocols/thread/tcat.rst:29]:
```suggestion
Currently, `BBTC Client`_ is the only available TCAT commissioner tool, and can be found in the OpenThread repository.
Refer to the tool's documentation for more information.
```
Then the following needs to be added to Line 100 of `links.txt`:
```
.. _`BBTC Client`: https://github.com/nrfconnect/sdk-openthread/tree/main/tools/tcat_ble_client
```

**Review by wiba-nordic — CHANGES_REQUESTED** [review]

File: doc/nrf/protocols/thread/tcat.rst (hunk 7 above)
Comment (canisLupus1313) [reply doc/nrf/protocols/thread/tcat.rst]: fixed.

File: doc/nrf/protocols/thread/tcat.rst (hunk 6 above)
Comment (canisLupus1313) [reply doc/nrf/protocols/thread/tcat.rst]: fixed.

File: doc/nrf/protocols/thread/tcat.rst (hunk 5 above)
Comment (canisLupus1313) [reply doc/nrf/protocols/thread/tcat.rst]: fixed.

File: doc/nrf/protocols/thread/tcat.rst [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+commissioning at scale makes it easy to wirelessly onboard Thread devices that are pre-installed in difficult-to-reach places, such as inside a ceiling or embedded in a wall.
+Instead of scanning physical install codes once installed devices are powered,
+Thread commissioning can be implemented over authenticated TLS (Transport Layer Security) by exchanging security certificates.
+This authenticated commissioning can be performed while in proximity, using a mobile device (phone, tablet) over a wireless connection such as a Bluetooth Low Energy radio.
+
+Building and Running
+********************
+
+Device site
+===========
```
[the remark this answers is not in the document]
Comment (canisLupus1313) [reply doc/nrf/protocols/thread/tcat.rst]: fixed.

File: doc/nrf/protocols/thread/tcat.rst [hunk 9]
```diff
@@ -0,0 +1,29 @@
+.. _thread_tcat:
+
+Thread Commissioning Over Authenticated TLS (TCAT)
+##################################################
+
+Developed to address the needs of professional installation and commercial building scenarios,
+commissioning at scale makes it easy to wirelessly onboard Thread devices that are pre-installed in difficult-to-reach places, such as inside a ceiling or embedded in a wall.
+Instead of scanning physical install codes once installed devices are powered,
+Thread commissioning can be implemented over authenticated TLS (Transport Layer Security) by exchanging security certificates.
+This authenticated commissioning can be performed while in proximity, using a mobile device (phone, tablet) over a wireless connection such as a Bluetooth Low Energy radio.
```
[the remark this answers is not in the document]
Comment (canisLupus1313) [reply doc/nrf/protocols/thread/tcat.rst]: fixed.

File: samples/openthread/cli/README.rst [hunk 10]
```diff
@@ -102,7 +102,7 @@ The following snippets are available:
 * ``multiprotocol`` - Enables Bluetooth LE support in this sample.
   Not compatible with the ``tcat`` snippet.
 * ``tcat`` - Enables support for Thread comissioning over authenticated TLS.
-  Not compatible with the ``multiprotocol`` snippet.
+  Not compatible with the ``multiprotocol`` snippet. For using TCAT please refer to :ref:`thread_tcat`
```
[the remark this answers is not in the document]
Comment (canisLupus1313) [reply samples/openthread/cli/README.rst]: fixed.

**Review by maciejbaczmanski — APPROVED** [review]

**Pull request merged**